src/ims.js

```diff
@@ -217,9 +219,13 @@ class Ims {
    *      other than `prod` or `stage` it is assumed to be the default
    *      value of `prod`. If not set, it will get the global cli env value. See https://github.com/adobe/aio-lib-env
    *      (which defaults to `prod` as well if not set)
+   * @param {ValidationCache} cache The cache instance to use.
    */
-  constructor (env = getCliEnv()) {
+  constructor (env = getCliEnv(), cache) {
```

Comment:
Why is the consumer of `Ims.js` required to pass the cache instance? Shouldn't this be internal to Ims module ?

Reply:
So I initially had this where `cache` was a simple boolean parameter on whether to cache validations or not. In that implementation, the class would internally initialize a cache with default parameters like you mentioned. I decided on this implementation so that developers could configure their own caches, and specifically decided on injecting a cache instance to avoid blowing up this constructor to include all the cache parameters

If we think cache customization is not necessary and that turning cache on or off is enough, then we could definitely simplify this back to a boolean. Wdyt?